lib/matplotlib/axes/_base.py

```diff
@@ -2430,6 +2430,12 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
                 dl.extend(y_finite)
 
             bb = mtransforms.BboxBase.union(dl)
+            vl = mtransforms.BboxBase.union([ax.viewLim for ax in shared])
```

Comment:
Optional: Could maybe go into the `if`.

Disadvantage: Would be duplicated for both ifs.
Advantage: Closer to the place of use. Gets only computed in the rare case that we need it (not that it would be a performance bottle neck.

Reply:
Put it in the `if` and will only get computed once!